hdr_html_canvas_element.md

```diff
@@ -0,0 +1,744 @@
+This application wishes to add HDR for UI effects or for special effects.
+
+In these examples, the application will want ``'extended'``, because that mode is guaranteed not to be disruptive to the application as it already exists.
+
+#### Example 2A: Brightening part of a UI to draw user's attention
+
+In this example, there exists a helpful piece of UI that the application wants to draw the user's attention to.
+That UI is being drawn by the canvas element.
+
+To draw the user's attention, the application pulses a doubling of the brightness of that part of the UI code.
+
+```xml
+  <html>
+  <body>
+  <canvas id='MyCanvas' width=500px height=500px
+          style='position:absolute; width: 100%; height:100%;'></canvas>
+  <script>
+    var canvas = document.getElementById('MyCanvas');
+    canvas.configureHDR('extended');
+    var context = canvas.getContext('2d', { precision: 'float16' });
+
+    function animateBrigtenedUI(timeInSections) {
+      // This code will brighten the content up to a factor of 2.
+      var brighteningFactor = 1 + Math.abs(Math.sin(timeInSeconds * 2*Math.PI));
+      context.filter = 'brightness(' + brighteningFactor + ')';
+
+      // And this code draws the regular SDR UI.
+      context.fillStyle = 'white';
+      context.fillRect(100, 100, 120, 20);
+      context.fillStyle = 'black';
+      context.fillText('I Am Some Helpful Text', 105, 115);
+    }
+  </script>
+  </body>
+  </html>
+```
+
+#### Example 2B: An effect in a WebGL application
+
+Consider an existing WebGL application with a lens flare effect.
+The lens flare is currently an 8-bit texture, but the application wants to have it actually be HDR, without changing any of the rest of application.
+
+Here is an outline of the previously existing application code. An existing HTMLCanvasElement, ``canvas`` is assumed to exist.
+
+```javascript
+    var gl = canvas.getContext('webgl2');
+
+    var myLensFlareData = new Uint8ClampedArray(4 * 256 * 256);
+    // Populate the data.
+
+    const tex = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, tex);
+    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, 256, 256, 0, gl.RGBA,
+                  gl.UNSIGNED_BYTE, myLensFlareData);
+    gl.useProgram(myLensFlareProgram);
+    gl.drawArrays(primitiveType, offset, count);
+  }
+```
+
+It is possible to enable just this one HDR texture, with minimal changes to the existing application.
+The comments in this code show the changes that are needed.
+
+```javascript
+    // Enable HDR for the canvas. Use the default mode of 'extended',
+    // because that will match the existing application behavior.
+    canvas.configureHDR('extended');
+
+    var gl = canvas.getContext('webgl2');
+
+    // Configure the swap chain to be floating-point. Leave the color space as
+    // the default of 'srgb'. This will leave all colors in [0, 1] unchanged,
+    // but allow for specifying colors outside of the range of [0, 1].
+    gl.configureSwapChain({ format: gl.RGBA16F });
```

Comment:
Needs updating per above comment about exact form of the WebGL API.

Reply:
Fixed